tests/cephadm/test_cluster_view.py

```diff
@@ -0,0 +1,108 @@
+    pass
+
+
+def verify_mon_service(host):
+    """
+    Verify MON service is up
+    Args:
+        host(ceph): Installer host
+    """
+    timeout, interval = 20, 6
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        out = CephAdm(host).ceph.orch.ls(service_type="mon", format="json")
+        configs = loads(out)
+        for config in configs:
+            running = config.get("status", {}).get("running")
+            size = config.get("status", {}).get("size")
+        if running == size:
+            log.info("All MON services are up and running")
+            return True
+        if w.expired:
+            raise ClusterViewError(
+                f"MON service failed: expected: {size}, running: {running}"
+            )
+
+
+def verify_mon_ips(host):
+    """
+    Verify MON IPs updated in ceph.conf
+    Args:
+        host(ceph): Installer host
+    """
+    # Get updated MON nodes
+    configs = loads(CephAdm(host).ceph.orch.ps(service_name="mon", format="json"))
+    mon_nodes = [config.get("daemon_id") for config in configs]
+
+    # Get MON node IPs
+    mon_ips = []
+    configs = loads(CephAdm(host).ceph.orch.host.ls(format="json"))
+    mon_ips = [c.get("addr") for c in configs if c.get("hostname") in mon_nodes]
+
+    # Validate MON IPs updated in ceph.conf
+    mon_ips_conf = host.remote_file(file_name=DEFAULT_CONF_PATH, file_mode="r")
+    mon_ips_conf = mon_ips_conf.readlines()[-1]
+    for mon_ip in mon_ips:
+        if mon_ip not in mon_ips_conf:
+            raise ClusterViewError("MON IPs not updated in ceph.conf")
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify admin node has updated view of cluster
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    install_node = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Apply MON placement
+    conf = {"pos_args": [], "placement": "5"}
+    out = CephAdm(install_node).ceph.orch.apply(service_name="mon", **conf)
+    if "Scheduled mon update" not in out:
+        raise ClusterViewError("Apply MON placement Failed")
+
+    verify_mon_service(install_node)
+    # Check Ceph health status
+    out = CephAdm(install_node).ceph.status()
+
+    # ls -lrt /etc/ceph/
+    dir_out = install_node.get_dir_list(dir_path=DEFAULT_CEPH_DIR)
+    if "ceph.conf" not in dir_out:
+        raise ClusterViewError(f"Ceph.conf not found in {DEFAULT_CEPH_DIR}")
+
+    # Verify MON IPs update in ceph.conf
+    verify_mon_ips(install_node)
+
+    # Get non-MON nodes
+    non_mon_nodes = []
+    nodes = ceph_cluster.get_nodes()
+    for node in nodes:
+        if node.role not in ["mon", "client"]:
+            non_mon_nodes.append(node)
+
+    # Apply MON on non OSD nodes
+    conf = {
+        "pos_args": [],
+        "host": f"\"{','.join([str(node) for node in non_mon_nodes])}\"",
+    }
+    out = CephAdm(install_node).ceph.orch.apply(service_name="mon", **conf)
+    if "Scheduled mon update" not in out:
```

Comment:
This statement fails, if you look at the log
```
cephadm shell ceph orch apply mon --host "<ceph.ceph.CephNode object at 0x7ff3ca097410>,<ceph.ceph.CephNode object at 0x7ff3c9fe8ed0>,<ceph.ceph.CephNode object at 0x7ff3ca055b90>,<ceph.ceph.CephNode object at 0x7ff3ca092cd0>,<ceph.ceph.CephNode object at 0x7ff3c93ea9d0>"``` 

it should be node.hostname.  In that case, the whole statement is wrong and is having no impact. Please revisit the logic

Reply:
I've corrected it now.